### ISSUE TITLE
when transforming .OWL files it requests .xml file

### ISSUE THREAD

**Issue by cstockloew** [universAAL/tools.eclipse-plugins#257]
_Originally Opened: @amedranogil (2013-01-11 10:30:15_)
_Originally Closed: 2013-01-18 11:12:47_

when I click on an owl file (i'm using this one: http://purl.obolibrary.org/obo/doid.owl ) the tool seems to do a first reading of the OWL and the second reading seems it is trying to read the .xml (non-existent).

Type... UML2
Reading...C:/Users/amedrano/workspaceOnt/ont.disease/Disease.owl
Read DONE C:/Users/amedrano/workspaceOnt/ont.disease/Disease.owl
Read base name.................http://purl.org/obo/owl
java.io.FileNotFoundException: C:\Users\amedrano\workspaceOnt\ont.disease\Disease.xml (El sistema no puede encontrar el archivo especificado)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.&lt;init&gt;(Unknown Source)
	at java.io.FileInputStream.&lt;init&gt;(Unknown Source)
	at sun.net.www.protocol.file.FileURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(Unknown Source)
	at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(Unknown Source)
	at org.apache.xerces.impl.XMLVersionDetector.determineDocVersion(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(Unknown Source)
	at org.universaal.tools.owl2uml.uml2.UML2Factory.readXML(UML2Factory.java:781)
	at org.universaal.tools.owl2uml.uml2.UML2Factory.&lt;init&gt;(UML2Factory.java:91)
	at org.universaal.tools.owl2uml.core.UML2Parser.loadOntology(UML2Parser.java:113)
	at org.universaal.tools.owl2uml.OWL2UML.main(OWL2UML.java:39)
	at org.universaal.tools.owl2uml.handlers.SampleHandler.execute(SampleHandler.java:79)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:293)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:476)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:508)
	at org.eclipse.ui.internal.handlers.HandlerService.executeCommand(HandlerService.java:169)
	at org.eclipse.ui.internal.handlers.SlaveHandlerService.executeCommand(SlaveHandlerService.java:241)
	at org.eclipse.ui.internal.handlers.SlaveHandlerService.executeCommand(SlaveHandlerService.java:241)
	at org.eclipse.ui.menus.CommandContributionItem.handleWidgetSelection(CommandContributionItem.java:829)
	at org.eclipse.ui.menus.CommandContributionItem.access$19(CommandContributionItem.java:815)
	at org.eclipse.ui.menus.CommandContributionItem$5.handleEvent(CommandContributionItem.java:805)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1053)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4165)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3754)
	at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
	at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
	at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
	at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:123)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:344)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:179)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:622)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:577)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1410)

--

From: _this issue has been automatically imported from our old issue tracker_

**Comment by cstockloew** [comment]
_Original author: Ioannis Moulos (2013-01-18 11:12:47)_

Because this is a universAALized version of (OWL2UML)tranformation, the tool is looking for  an .xml file in order to include the additional info to the generated .uml file (package name, abstract classes etc.). Samples of such .xml files can be found in the &quot;samples&quot; folder(http://forge.universaal.org/svn/uaaltools/trunk/transformations/org.universaal.tools.owl2uml/samples)

**Comment by cstockloew** [comment]
_Original author: @amedranogil (2013-01-18 11:32:00)_

may I add a feature request for a wizard/form to easily create such metadata XMLs?

**Comment by cstockloew** [comment]
_Original author: Ioannis Moulos (2013-01-21 12:35:29)_

Also we updated the tool to support URIs with &quot;/&quot; instead of &quot;#&quot; as class prefixes.

Regarding the wizard, we are currently out of resources on WP3 but the task can be implemented easily by any partner since we have provided the corresponding .xsd file. We believe it is not critical to have such a tool since the .xml file is simple and can be editted with notepad.
